### PR TITLE
alert "warning" added

### PR DESCRIPTION
**Pull request by LucynaMaliszewska**
This is a feature that gives possibility to add warning about some changes in data cubes e.g inconsistent data or lack of data. With this alert you can give some important information about data cube.

### CODE REVIEW

File: src/common/models/data-cube/data-cube.ts [hunk 1]
```diff
@@ -473,9 +487,11 @@ export class DataCube implements Instance<DataCubeValue, DataCubeJS> {
       source: this.source,
       dimensions: this.dimensions.toJS(),
       measures: this.measures.toJS(),
-      refreshRule: this.refreshRule.toJS()
+      refreshRule: this.refreshRule.toJS(),
+      warningTitle: this.warningTitle,
```
Comment (adrianmroz) [inline src/common/models/data-cube/data-cube.ts]: `warningTitle` is optional, `if` is needed

File: src/common/models/data-cube/data-cube.ts [hunk 2]
```diff
@@ -561,6 +579,23 @@ export class DataCube implements Instance<DataCubeValue, DataCubeJS> {
     };
   }
 
+  private parseWarningTitle({ warningTitle, warningDescription }: DataCubeValue): { warningTitle: string, warningDescription?: string } {
```
Comment (adrianmroz) [inline src/common/models/data-cube/data-cube.ts]: Looks like `parseDescription`, maybe reuse?

File: src/client/components/info-bubble/info-bubble.tsx [hunk 3]
```diff
@@ -23,7 +23,8 @@ import { SvgIcon } from "../svg-icon/svg-icon";
 import "./info-bubble.scss";
```
Comment (adrianmroz) [inline src/client/components/info-bubble/info-bubble.tsx]: You're making this component dependant on some edge case (warnings presented in header). You should make specialised version of this component via composition.

File: src/client/views/cube-view/cube-header-bar/cube-header-bar.tsx [hunk 4]
```diff
@@ -262,6 +262,7 @@ export class CubeHeaderBar extends React.Component<CubeHeaderBarProps, CubeHeade
       </div>
       <div className="title" onClick={onNavClick}>{dataCube.title}</div>
       {dataCube.description && <InfoBubble className="cube-description" description={dataCube.description}/>}
+      {(dataCube.warningTitle || dataCube.warningDescription) && <InfoBubble  className="cube-description" icon="warning" description={dataCube.warningDescription} title={dataCube.warningTitle}/> }
```
Comment (adrianmroz) [inline src/client/views/cube-view/cube-header-bar/cube-header-bar.tsx]: Why warning is presented only here? Why not on data cube list view?

File: src/common/models/data-cube/data-cube.ts [hunk 5]
```diff
@@ -339,7 +343,9 @@ export class DataCube implements Instance<DataCubeValue, DataCubeJS> {
       defaultPinnedDimensions: parameters.defaultPinnedDimensions ? OrderedSet(parameters.defaultPinnedDimensions) : null,
       maxSplits: parameters.maxSplits,
       maxQueries: parameters.maxQueries,
-      refreshRule
+      refreshRule,
+      warningTitle: parameters.warningTitle,
+      warningDescription: parameters.warningDescription,
```
Comment (mkuthan) [inline src/common/models/data-cube/data-cube.ts]: Fields warningTitle and warningDescription are duplicates of title and description. If Turnilo administrator has to add some additional information to data cube he/she should edit existing title/description.

If additional restrictions are really needed we should consider boolean flags like: enabled: [true|false] (true by default to be backward compatible).

With the proposed design we end up with: warningTitle, errorTitle, holidayTitle, etc ;)

**Comment by adrianmroz** [comment]
Reopen please  after requested changes

**Pull request closed without merging**